(l2) streams_csv_store

oceandlr opened the pull request:
streams_csv_store (was l2_streams_csv_store).

oceandlr commented:
This is the revamp of #654

oceandlr commented:
Hold off on this. I will rename streams to stream, as per the related discussion.

oceandlr commented:
1) As per @baallan separate branch 00fb306 is in ldmsd.c. This should be a separate commit
2) I have taken 65e9073 which is the test as a cherry-pick and changed to ldmsd_stream_publish to have this correct with the deprecation. You will need to add your disconnect test to ldmsd_stream_publish.c instead of hello_cat_publisher which is now deprecated.
3) I have taken some of 3469953, but not all. Some of these are not consistent with 
store_csv.c
I

oceandlr commented:
Meeting with @baallan on 8/31. Goal is to be able to take this after that meeting.

baallan commented:
@oceandlr the deprecation/renaming should have targeted subdir stream_csv instead of just stream/ which is ambiguous.

oceandlr commented:
@baallan Deliberately so. This way we can put other streams stuff under the same subdir.
I expect in the future that since the stream_cb is really the same as the normal store functionality, that we can evolve to a design where streams stores go away and the normal stores support both.

tom95858 commented:
@oceandlr, @baallan where are we on this pull request?

oceandlr commented:
@tom95858 I will put the various termination items in the locations as in the Discussion thread 785 Plugin End state this weekend. I will rebase and let you know when you can merge.

Sorry about the delay.

baallan commented:
@tom95858 I have a test script set for this plugin; I don't think Ann is planning to include it in the revised pull request, so I'll keep an eye out and rerun that when this is updated.

tom95858 commented:
@oceandlr What's the status of this? It currently has merge conflicts.

oceandlr commented:
I have not had a chance to update since 10 days ago. I will let you know when it is ready. Sorry about the delay.

tom95858 commented:
@oceandlr ***@***.***> no worries. I was just trying to clean
up the pull request list in advance of testing on Voltrino

On Mon, Sep 27, 2021 at 12:15 PM oceandlr ***@***.***> wrote:

> I have not had a chance to update since 10 days ago. I will let you know
> when it is ready. Sorry about the delay.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ovis-hpc/ovis/pull/777#issuecomment-928151642>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABVTPXFWREGPA2B5PXCU7DTUECYCZANCNFSM5CSLOKAQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 
Thomas Tucker, President, Open Grid Computing, Inc.

oceandlr commented:
@tom95858 I have put the items from fini into term and also used the function that you said is effectively stream unsubscribe. Please take this now.
I have also removed some of the various hello utilities that are now replaced by ldmsd_stream_publish @nichamon 

@valleydlr

baallan commented:
@oceandlr there appears to still be a deadlock in the code. I'm checking on what the smallest fix is.
@tom95858 Are we accepting code with c++ line-end comments? If so can you update the developer guidelines accordingly?

oceandlr commented:
what are you testing and what is happening @baallan

baallan commented:
I am testing the daemon shutting down while messages are still coming in. The daemon is hanging instead of exiting, which was something I saw in earlier testing until I fixed it in my 2nd review branch.  I'm working with gdb now to verify its the same problem and not a new one.

baallan commented:
@oceandlr I have opened a new pull request # 3 on your fork with the non-cosmetic changes needed (unless @tom95858 has a better solution). The deadlock in play has gdb traces included with the pull request.

tom95858 commented:
> @tom95858 Are we accepting code with c++ line-end comments? If so can you update the developer guidelines accordingly?

We should refactor the // to /* */ comments to be consistent with the coding guidelines. Also there is code that is commented out which should be removed.
